### PR TITLE
[sanity_check] Use golden config to reboot for `adaptive_recover`

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205
- [x] 202305

#### What is the motivation for this PR?
For `adaptive_recover`, it will reboot if `monit` has failures. But the DUT keeps using the `config_db.json`, which might be corrupted by previous tests in nightly.
Let's reboot with the running golden config for the reboot in `adaptive_recover`.

#### How did you do it?
Override `config_db.json` with the running golden config before reboot.

#### How did you verify/test it?
Run on dualtor testbed with monit service `routeCheck` fails and `config_db.json` is corrupted with mux config `manual`. 
Verify after reboot, monit is recovered and mux config is back to `auto`.

### CODE REVIEW

File: tests/common/plugins/sanity_check/recover.py [hunk 1]
```diff
@@ -23,6 +23,15 @@ def reboot_dut(dut, localhost, cmd):
     else:
         reboot_type = REBOOT_TYPE_COLD
 
+    if reboot_with_running_golden_config:
+        logging.info("Reboot DUT with the running golden config")
+        dut.copy(
+            src="/etc/sonic/running_golden_config.json",
```
Comment (zjswhhh) [inline tests/common/plugins/sanity_check/recover.py:29]: Does the src file always exist? Do we need to add checker?

Comment (xwjiang-ms) [reply tests/common/plugins/sanity_check/recover.py:29]: Agree, suggest to copy golden config with best effort, if no src file, print a warning msg

Comment (lolyu) [reply tests/common/plugins/sanity_check/recover.py]: Updated

**Review by xwjiang-ms — APPROVED** [review]
LGTM

**Review by wangxin — APPROVED** [review]

**Pull request merged**